### ISSUE TITLE
Bad sad error gets thrown when compiling AWS S3 module

### ISSUE THREAD

**Issue by miyurud** [ballerina-platform/ballerina-lang#26591]
**Description:**
I get the following bad sad error when building `module-ballerinax-aws.s3` with Ballerina slp5.

```
[2020-10-26 11:19:37,360] SEVERE {b7a.log.crash} - class io.ballerina.compiler.internal.syntax.ExternalTreeNodeList cannot be cast to class io.ballerina.compiler.syntax.tree.MetadataNode (io.ballerina.compiler.internal.syntax.ExternalTreeNodeList and io.ballerina.compiler.syntax.tree.MetadataNode are in unnamed module of loader 'app') 
java.lang.ClassCastException: class io.ballerina.compiler.internal.syntax.ExternalTreeNodeList cannot be cast to class io.ballerina.compiler.syntax.tree.MetadataNode (io.ballerina.compiler.internal.syntax.ExternalTreeNodeList and io.ballerina.compiler.syntax.tree.MetadataNode are in unnamed module of loader 'app')
	at java.base/java.util.Optional.map(Optional.java:265)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.getAnnotations(BLangNodeTransformer.java:480)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:549)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:436)
	at io.ballerina.compiler.syntax.tree.ModuleVariableDeclarationNode.apply(ModuleVariableDeclarationNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:530)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:436)
	at io.ballerina.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:460)
	at org.wso2.ballerinalang.compiler.parser.Parser.generateCompilationUnitNew(Parser.java:121)
	at org.wso2.ballerinalang.compiler.parser.Parser.parse(Parser.java:94)
	at org.wso2.ballerinalang.compiler.PackageLoader.parse(PackageLoader.java:487)
	at org.wso2.ballerinalang.compiler.PackageLoader.loadEntryPackage(PackageLoader.java:374)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:166)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:147)
	at org.ballerinalang.packerina.task.CompileTask.execute(CompileTask.java:66)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:435)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at org.ballerinalang.tool.Main.main(Main.java:57)
```

**Steps to reproduce:**
Download Ballerina slp5 release, clone the latest [module-ballerinax-aws.s3](https://github.com/ballerina-platform/module-ballerinax-aws.s3) and issue `ballerina build -a -c`

**Affected Versions:**
Ballerina slp5

**OS, DB, other environment details and versions:**
Ubuntu 18.04

**Related Issues (optional):**
N/A

**Suggested Labels (optional):**
N/A

**Suggested Assignees (optional):**
N/A

**Comment by lochana-chathura** [comment]
Root cause for the CCE is fixed with PR #27328.

But, one should not get above log unless the module contains a bal file with invalid top level construct.

**Comment by rdhananjaya** [comment]
Issue seems to be fixed as of Swan Lake beta-1
